mypy/plugins/dataclasses.py

```diff
@@ -184,7 +190,11 @@ def reset_init_only_vars(self, info: TypeInfo, attributes: List[DataclassAttribu
         """Remove init-only vars from the class and reset init var declarations."""
         for attr in attributes:
             if attr.is_init_var:
-                del info.names[attr.name]
+                try:
+                    del info.names[attr.name]
+                except KeyError:
```

Comment:
Similar to above, I'd slightly prefer an if statement instead of try/except here.

Reply:
Fixed.